core/src/main/java/org/bitcoinj/core/Transaction.java

```diff
@@ -460,16 +460,34 @@ public void setUpdateTime(Date updatedAt) {
     /**
      * These constants are a part of a scriptSig signature on the inputs. They define the details of how a
      * transaction can be redeemed, specifically, they control how the hash of the transaction is calculated.
-     * <p/>
-     * In Bitcoin Core, this enum also has another flag, SIGHASH_ANYONECANPAY. In this implementation,
-     * that's kept separate. Only SIGHASH_ALL is actually used in Bitcoin Core today. The other flags
-     * exist to allow for distributed contracts.
      */
     public enum SigHash {
-        ALL,         // 1
-        NONE,        // 2
-        SINGLE,      // 3
+        ALL(1),
+        NONE(2),
+        SINGLE(3),
+        ANYONECANPAY(0x80), // Caution: Using this type in isolation is non-standard. Treated similar to ANYONECANPAY_ALL.
+        ANYONECANPAY_ALL(0x81),
+        ANYONECANPAY_NONE(0x82),
+        ANYONECANPAY_SINGLE(0x83),
+        UNSET(0); // Caution: Using this type in isolation is non-standard. Treated similar to ALL.
+
+        public final int type;
+
+        /**
+         * @param type
+         */
+        private SigHash(final int type) {
+            this.type = type;
+        }
+
+        /**
+         * @return the type as a byte
+         */
+        public byte byteValue() {
+            return (byte) this.type;
+        }
     }
+
     public static final byte SIGHASH_ANYONECANPAY_VALUE = (byte) 0x80;
```

Comment:
on second thought, i remember that left that around because it was a public member that existing code could depend on. i think i'll leave it as a public member, but remove references to it. it effectively becomes deprecated. i'll probably add a comment that it's now deprecated and to use the enum.